plugins/ocm-backend/src/helpers/parser.ts

```diff
@@ -58,3 +60,41 @@ export const parseManagedCluster = (cluster: any): ClusterDetails => {
     ...parsedClusterInfo,
   };
 };
+
+export const parseUpdateInfo = (clusterInfo: any) => {
+  const versionAvailableUpdates =
+    clusterInfo.status.distributionInfo.ocp.versionAvailableUpdates;
+  const versionsAvailable =
+    clusterInfo.status.distributionInfo.ocp.availableUpdates;
+  const version = versionsAvailable
+    ? maxSatisfying(versionsAvailable, '*')
+    : undefined;
+  const url = versionAvailableUpdates
+    ? versionAvailableUpdates[versionsAvailable.indexOf(version)].url
+    : undefined;
+
+  return {
+    update: {
+      available: versionsAvailable ? versionsAvailable.length !== 0 : false,
```

Comment:
We knew this much earlier. Why don't we return right away if there are no updates? Instead you condition on the `versionsAvailable` 3 times (lines 69,73 and here).

Reply:
Resolved in the lastest commit.